add koonovo 3Dprinter

KOONOVO3DPrinter opened the pull request.

konskarm commented on resources/definitions/koonovo_base.def.json:
```diff
@@ -0,0 +1,136 @@
+{
+    "name": "Creawsome Base Printer",
```
You probably want to change that to "Koonovo Base Printer"

konskarm reviewed (changes requested):
Also please move all qualities inside a folder named koonovo.

KOONOVO3DPrinter commented:
OK，thanks
&nbsp;
&nbsp;
------------------&nbsp;Original&nbsp;------------------
From: "Konstantinos Karmas"; 
Date: 2020年9月24日(星期四) 下午3:33
To: "Ultimaker/Cura"; 
Cc: "KOONOVO3DPrinter"; "Author"; 
Subject: Re: [Ultimaker/Cura] add koonovo 3Dprinter (#8433)

&nbsp;


 
@konskarm requested changes on this pull request.
 
Also please move all qualities inside a folder named koonovo.
 
—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or unsubscribe.

KOONOVO3DPrinter commented:
> 另外，请将所有品质移动到名为koonovo的文件夹中。
Ok, it has been changed

fieldOfView commented:
I may be nitpicking, but don't you have your own logo in a better quality? This is part of the image at 100% magnification:
![image](https://user-images.githubusercontent.com/143551/94267324-ad11ce00-ff3b-11ea-99be-421cdc5c4989.png)

KOONOVO3DPrinter commented:
Ok, thank you for your suggestion. I will try my best to replace it with a better picture


&nbsp;
&nbsp;
------------------&nbsp;Original&nbsp;------------------
From: "Aldo Hoeben"; 
Date: 2020年9月25日(星期五) 晚上8:31
To: "Ultimaker/Cura"; 
Cc: "KOONOVO3DPrinter"; "Author"; 
Subject: Re: [Ultimaker/Cura] add koonovo 3Dprinter (#8433)

&nbsp;


 
I may be nitpicking, but don't you have your own logo in a better quality? This is part of the image at 100% magnification:
 
 
—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or unsubscribe.

KOONOVO3DPrinter commented:
> 另外，请将所有品质移动到名为koonovo的文件夹中。
Ok, the KOONOVO file has been established and the quality file has been moved into it, other files have been verified, when can it be incorporated into the new version of CURA, thank you

KOONOVO3DPrinter commented:
> 另外，请将所有品质移动到名为koonovo的文件夹中。
Changed, request merged into CURA

KOONOVO3DPrinter commented:
This request has been modified and improved, request added to CURA, thank you
&nbsp;
&nbsp;
------------------&nbsp;Original&nbsp;------------------
From: "Konstantinos Karmas"; 
Date: 2020年9月24日(星期四) 下午3:33
To: "Ultimaker/Cura"; 
Cc: "KOONOVO3DPrinter"; "Author"; 
Subject: Re: [Ultimaker/Cura] add koonovo 3Dprinter (#8433)

&nbsp;


 
@konskarm requested changes on this pull request.
 
Also please move all qualities inside a folder named koonovo.
 
—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or unsubscribe.

konskarm reviewed (changes requested):
Please rename all global quality profiles to start with "koonovo_".
E.g. resources/quality/koonovo/base_global_high.inst.cfg should be named resources/quality/koonovo/koonovo_base_global_high.inst.cfg.

Also I see that the size of koonovo_elf.def.json is 310x310x345mm, while the specifications of the printer list it as 300x300x330mm.
Similarly, the koonovo_pyramid.def.json is 310x310x400mm while in the specification it is mentioned as 300x300x400mm. Can you validate whether this is correct?

KOONOVO3DPrinter commented:
Thanks for reminding me. I have renamed it. I am sure the machine is the correct size. The dimensions of Koonovo_ELF 300x300x330 and koonovo_pyramid 300x300x400 are the basic dimensions of the machine. 310x310x345 and 3100x310x400 are the maximum print size of the machine, we need to set the maximum size when setting the machine. I am the designer of these machines, so I can guarantee that it is correct. Please help add it to the next version of CURA, thank you!
&nbsp;
------------------&nbsp;Original&nbsp;------------------
From: &nbsp;"Konstantinos Karmas"<notifications@github.com&gt;;
Date: &nbsp;Mon, Oct 5, 2020 08:16 PM
To: &nbsp;"Ultimaker/Cura"<Cura@noreply.github.com&gt;; 
Cc: &nbsp;"KOONOVO3DPrinter"<engineer@koonovo.com&gt;; "Author"<author@noreply.github.com&gt;; 
Subject: &nbsp;Re: [Ultimaker/Cura] add koonovo 3Dprinter (#8433)

&nbsp;



 
@konskarm requested changes on this pull request.
 
Please rename all global quality profiles to start with "koonovo_".
 E.g. resources/quality/koonovo/base_global_high.inst.cfg should be named resources/quality/koonovo/koonovo_base_global_high.inst.cfg.
 
Also I see that the size of koonovo_elf.def.json is 310x310x345mm, while the specifications of the printer list it as 300x300x330mm.
 Similarly, the koonovo_pyramid.def.json is 310x310x400mm while in the specification it is mentioned as 300x300x400mm. Can you validate whether this is correct?
 
—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or unsubscribe.

konskarm commented on resources/definitions/koonovo_base.def.json:
```diff
@@ -0,0 +1,136 @@
+	  "platform": "koonovo.obj",
+        "platform_texture": "koonovo.png",
+
+        "first_start_actions": [ "DiscoverUM3Action" ],
```
```suggestion
```
This is only for printers that are discoverable via network.

konskarm replied:
This is the final minor issue that I see

KOONOVO3DPrinter commented:
Okay, thank you. I'll change it tomorrow.




---Original---
From: "Konstantinos Karmas"<notifications@github.com&gt;
Date: Tue, Oct 6, 2020 22:15 PM
To: "Ultimaker/Cura"<Cura@noreply.github.com&gt;;
Cc: "Author"<author@noreply.github.com&gt;;"KOONOVO3DPrinter"<engineer@koonovo.com&gt;;
Subject: Re: [Ultimaker/Cura] add koonovo 3Dprinter (#8433)




 
@konskarm commented on this pull request.
 
 
In resources/definitions/koonovo_base.def.json:
 &gt; @@ -0,0 +1,136 @@ +{ +    "name": "KOONOVO Base Printer", +    "version": 2, +    "inherits": "fdmprinter", +    "metadata": { +        "visible": false, +        "author": "KOONOVO", +        "manufacturer": "KOONOVO", +        "file_formats": "text/x-gcode", +  "platform": "koonovo.obj", +        "platform_texture": "koonovo.png", + +        "first_start_actions": [ "DiscoverUM3Action" ],  
This is the final minor issue that I see
 
—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or unsubscribe.

KOONOVO3DPrinter commented:
Hello, the "first_start_actions" you said: ["DiscoverUM3Action"], I have changed it and added a new type of our machine, this is a separate double X axis, double extrusion machine, I have verified this setting is reasonable, can you help me to add it to the next version?&nbsp;
&nbsp;
------------------&nbsp;Original&nbsp;------------------
From: "Konstantinos Karmas"; 
Date: 2020年10月6日(星期二) 晚上10:15
To: "Ultimaker/Cura"; 
Cc: "KOONOVO3DPrinter"; "Author"; 
Subject: Re: [Ultimaker/Cura] add koonovo 3Dprinter (#8433)

&nbsp;


 
@konskarm commented on this pull request.
 
 
In resources/definitions/koonovo_base.def.json:
 &gt; @@ -0,0 +1,136 @@ +{ +    "name": "KOONOVO Base Printer", +    "version": 2, +    "inherits": "fdmprinter", +    "metadata": { +        "visible": false, +        "author": "KOONOVO", +        "manufacturer": "KOONOVO", +        "file_formats": "text/x-gcode", +	  "platform": "koonovo.obj", +        "platform_texture": "koonovo.png", + +        "first_start_actions": [ "DiscoverUM3Action" ],  
This is the final minor issue that I see
 
—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or unsubscribe.

Ghostkeeper commented:
The tests are currently failing because `koonovo_kn3_extruder_0.def.json` and `koonovo_kn3_extruder_1.def.json` are referring to a printer called `KOONOVO` which doesn't exist. They both have this line:

    "machine": "KOONOVO",

I think you'd want to refer to that kn3 printer:

    "machine": "koonovo_kn3",

Other than that the printer looks good to me from a review (I haven't tried loading it up in Cura yet with a test).

KOONOVO3DPrinter commented:
Thank you. I have changed it but still got an error, but it is ok in my test. What is the reason
&nbsp;
&nbsp;
------------------&nbsp;Original&nbsp;------------------
From: "Ghostkeeper"; 
Date: 2020年10月14日(星期三) 下午5:32
To: "Ultimaker/Cura"; 
Cc: "KOONOVO3DPrinter"; "Author"; 
Subject: Re: [Ultimaker/Cura] add koonovo 3Dprinter (#8433)

&nbsp;


 
The tests are currently failing because koonovo_kn3_extruder_0.def.json and koonovo_kn3_extruder_1.def.json are referring to a printer called KOONOVO which doesn't exist. They both have this line:
 "machine": "KOONOVO",  
I think you'd want to refer to that kn3 printer:
 "machine": "koonovo_kn3",  
Other than that the printer looks good to me from a review (I haven't tried loading it up in Cura yet with a test).
 
—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or unsubscribe.

konskarm commented:
Currently the build system is failing due to an internal issue. It's something that we are working to fix.

Ghostkeeper commented:
Well it's also failing on an actual mistake. The machine it's referring to is the filename of the printer definition but without the extension, so `koonovo_kn3`. But you've written `KOONOVO_KN3`. Like most things in programming, this is case sensitive.

KOONOVO3DPrinter commented:
Nín shuō de shì koonovo_kn3_extruder_0.Def.Json/koonovo_kn3_extruder_0.Def.Json  "machine": "KOONOVO_KN3" ? Wǒ gēnggǎile, xiànzài biānyì tōngguòle qǐng nín tiānjiā jìn CURA de xià yī bǎnběn
展开




123/5000












Are you talking about koonovo_kn3_extruder_0.def.json/koonovo_kn3_extruder_0.def.json "machine": "KOONOVO_KN3"? I changed it, now it’s compiled, please add it to the next version of CURA





&nbsp;
&nbsp;
------------------&nbsp;Original&nbsp;------------------
From: "Ghostkeeper"; 
Date: 2020年10月15日(星期四) 晚上10:30
To: "Ultimaker/Cura"; 
Cc: "KOONOVO3DPrinter"; "Author"; 
Subject: Re: [Ultimaker/Cura] add koonovo 3Dprinter (#8433)

&nbsp;


 
Well it's also failing on an actual mistake. The machine it's referring to is the filename of the printer definition but without the extension, so koonovo_kn3. But you've written KOONOVO_KN3. Like most things in programming, this is case sensitive.
 
—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or unsubscribe.

Ghostkeeper commented:
What language is that? Vietnamese? They have so many accents on their letters. I can't understand it, sorry.

KOONOVO3DPrinter commented:
Sorry, I mean I have changed, please add the machine to the next version, thank you very much
&nbsp;
&nbsp;
------------------&nbsp;Original&nbsp;------------------
From: "Ghostkeeper"; 
Date: 2020年10月19日(星期一) 晚上9:45
To: "Ultimaker/Cura"; 
Cc: "KOONOVO3DPrinter"; "Author"; 
Subject: Re: [Ultimaker/Cura] add koonovo 3Dprinter (#8433)

&nbsp;


 
What language is that? Vietnamese? They have so many accents on their letters. I can't understand it, sorry.
 
—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or unsubscribe.

KOONOVO3DPrinter commented:
Hello, I recently added a 420*420*400 independent 3D printer with dual X-axis motion. It can realize two-color printing, printing at the same time, and mirror printing; Can you help me add it to the latest version? Thank you
&nbsp;
&nbsp;
------------------&nbsp;Original&nbsp;------------------
From: "Ghostkeeper"; 
Date: 2020年10月19日(星期一) 晚上9:45
To: "Ultimaker/Cura"; 
Cc: "KOONOVO3DPrinter"; "Author"; 
Subject: Re: [Ultimaker/Cura] add koonovo 3Dprinter (#8433)

&nbsp;


 
What language is that? Vietnamese? They have so many accents on their letters. I can't understand it, sorry.
 
—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or unsubscribe.

Ghostkeeper commented on resources/definitions/koonovo_elf.def.json:
```diff
@@ -0,0 +1,26 @@
+        "machine_name": { "default_value": "KOONOVO ELF" },
+        "machine_width": { "default_value": 310 },
+        "machine_depth": { "default_value": 310 },
+        "machine_height": { "default_value": 345 },
```
Alibaba lists the specs of this printer as being 300x300x330: https://www.alibaba.com/product-detail/3d-Desktop-High-precision-3D-Printer_62404382636.html?spm=a2700.galleryofferlist.normal_offer.d_title.bae94c94i7W5gy

Ghostkeeper replied:
Alibaba lists the specs of this printer as being 300x300x330: https://www.alibaba.com/product-detail/3d-Desktop-High-precision-3D-Printer_62404382636.html?spm=a2700.galleryofferlist.normal_offer.d_title.bae94c94i7W5gy

Ghostkeeper commented on resources/definitions/koonovo_kn3.def.json:
```diff
@@ -0,0 +1,160 @@
+{
+    "name": "KOONOVO KN3 IDEX",
```
I was unable to find the KN3 or KN5 online. Are you sure the names are correct? Do you have a spec sheet anywhere?

Ghostkeeper commented on resources/definitions/koonovo_kn5.def.json:
```diff
@@ -0,0 +1,159 @@
+{
+    "name": "KOONOVO KN5 IDEX",
```
I was unable to find the KN3 or KN5 online. Are you sure the names are correct? Do you have a spec sheet anywhere?

Ghostkeeper commented on resources/definitions/koonovo_pyramid.def.json:
```diff
@@ -0,0 +1,26 @@
+        "machine_name": { "default_value": "KOONOVO Pyramid" },
+        "machine_width": { "default_value": 310 },
+        "machine_depth": { "default_value": 310 },
+        "machine_height": { "default_value": 400 },
```
Alibaba lists the size of this printer as 300x300x400: https://www.alibaba.com/product-detail/3d-Desktop-3D-Printer-Kits-KOONOVO_62404380206.html?spm=a2700.galleryofferlist.normal_offer.d_title.39704c94hBUbHe

Ghostkeeper reviewed (changes requested):
We'd prefer to have printer names and manufacturers use normal casing instead of stylized capital letters, unless it is an abbreviation (like FFF).

KOONOVO3DPrinter commented:
Thank you for reminding me. I'm sure this machine is the right size. The dimensions of Koonovo_ELF 300x300x330 and Koonovo_pyramid 300x300x400 are the basic print dimensions of the machine. 310x310x345 and 3100x310x400 are the maximum printing size of the machine. We need to set the maximum size when setting the machine. I am the designer of these machines, so I can guarantee that it is correct. KN3 and KN5 make our latest machine to be sold on the Internet, so there is no direct information yet. Since it is a bit troublesome to set parameters for the independent double X-axis machine, I hope the user can use CURA slice directly after receiving the machine. I hope you can add it to the new CURA, thank you very much!&nbsp;
&nbsp;
------------------&nbsp;Original&nbsp;------------------
From: "Ghostkeeper"; 
Date: 2020年10月27日(星期二) 凌晨1:33
To: "Ultimaker/Cura"; 
Cc: "KOONOVO3DPrinter"; "Author"; 
Subject: Re: [Ultimaker/Cura] add koonovo 3Dprinter (#8433)

&nbsp;


 
@Ghostkeeper requested changes on this pull request.
 
We'd prefer to have printer names and manufacturers use normal casing instead of stylized capital letters, unless it is an abbreviation (like FFF).

 
In resources/definitions/koonovo_kn3.def.json:
 &gt; @@ -0,0 +1,160 @@ +{ +    "name": "KOONOVO KN3 IDEX",  
I was unable to find the KN3 or KN5 online. Are you sure the names are correct? Do you have a spec sheet anywhere?
 
 
In resources/definitions/koonovo_kn5.def.json:
 &gt; @@ -0,0 +1,159 @@ +{ +    "name": "KOONOVO KN5 IDEX",  
I was unable to find the KN3 or KN5 online. Are you sure the names are correct? Do you have a spec sheet anywhere?
 
 
In resources/definitions/koonovo_elf.def.json:
 &gt; @@ -0,0 +1,26 @@ +{ +    "name": "KOONOVO ELF", +    "version": 2, +    "inherits": "koonovo_base", +    "overrides": { +        "machine_name": { "default_value": "KOONOVO ELF" }, +        "machine_width": { "default_value": 310 }, +        "machine_depth": { "default_value": 310 }, +        "machine_height": { "default_value": 345 },  
Alibaba lists the specs of this printer as being 300x300x330: https://www.alibaba.com/product-detail/3d-Desktop-High-precision-3D-Printer_62404382636.html?spm=a2700.galleryofferlist.normal_offer.d_title.bae94c94i7W5gy
 
 
In resources/definitions/koonovo_pyramid.def.json:
 &gt; @@ -0,0 +1,26 @@ +{ +    "name": "KOONOVO Pyramid", +    "version": 2, +    "inherits": "koonovo_base", +    "overrides": { +        "machine_name": { "default_value": "KOONOVO Pyramid" }, +        "machine_width": { "default_value": 310 }, +        "machine_depth": { "default_value": 310 }, +        "machine_height": { "default_value": 400 },  
Alibaba lists the size of this printer as 300x300x400: https://www.alibaba.com/product-detail/3d-Desktop-3D-Printer-Kits-KOONOVO_62404380206.html?spm=a2700.galleryofferlist.normal_offer.d_title.39704c94hBUbHe
 
—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or unsubscribe.

Ghostkeeper commented:
Ah, these have been merged here: https://github.com/Ultimaker/Cura/commit/77f5fefc5da9ad8f91afd4174ca169abdaea06cc

Only Github doesn't recognise it as merging this pull request since I squashed the commits. I normally squash the commits if people create an excessive amount of commits without any descriptive commit messages.
It'll be included in Cura 4.9! Thank you for your contribution.

KOONOVO3DPrinter commented:
thank you very much
&nbsp;
&nbsp;
------------------&nbsp;Original&nbsp;------------------
From: "Ghostkeeper"; 
Date: 2020年10月29日(星期四) 凌晨2:04
To: "Ultimaker/Cura"; 
Cc: "KOONOVO3DPrinter"; "Author"; 
Subject: Re: [Ultimaker/Cura] add koonovo 3Dprinter (#8433)

&nbsp;


 
Ah, these have been merged here: 77f5fef
 
Only Github doesn't recognise it as merging this pull request since I squashed the commits. I normally squash the commits if people create an excessive amount of commits without any descriptive commit messages.
 It'll be included in Cura 4.9! Thank you for your contribution.
 
—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or unsubscribe.
